lib/src/visitor/async_evaluate.dart

```diff
@@ -1987,9 +1993,13 @@ class _EvaluateVisitor
   Future<Value?> visitWarnRule(WarnRule node) async {
     var value =
         await _addExceptionSpanAsync(node, () => node.expression.accept(this));
-    _logger.warn(
-        value is SassString ? value.text : _serialize(value, node.expression),
-        trace: _stackTrace(node.span));
+    var message =
+        value is SassString ? value.text : _serialize(value, node.expression);
+
+    // Don't use [_warn] because we don't want to pass the span to the logger.
+    if (_warningsEmitted.add(Tuple2(message, node.span))) {
+      _warn(message, node.span);
```

Comment:
Should this be `_logger.warn()`?

Reply:
the comment above makes me thing it should